File: backend/src/models/project.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class Project(BaseModel):
    id: Optional[int] = None
    nombre: str
    description: Optional[str] = None
    user_id: int
    created_at: Optional[datetime] = None

    def can_be_deleted(self) -> bool:
        """Business logic for project deletion"""
        return True

class ProjectCreate(BaseModel):
    nombre: str
    description: Optional[str] = None

File: frontend/src/lib/models/project.ts
export interface Project {
  id?: number;
  nombre: string;
  description?: string;
  user_id: number;
  created_at?: Date;
}

export interface ProjectCreate {
  nombre: string;
  description?: string;
}

export function canBeDeleted(project: Project): boolean {
  return true;
}
